action.yml

```
@@ -0,0 +1,76 @@
+name: Mend Scan
+description: Scan for vulnerabilities and license compliance
+
+inputs:
+  api-key:
+    required: true
+    description: A unique identifier of your organization. Used to identify the organization in plugins.
+
+  product-token:
+    required: true
+    description: Unique identifier of the product to update; found on the Integrate page. Required if Project Token is not defined.
+
+  check-policies:
+    required: false
+    default: true
+    description: Whether to send the check policies request before updating mend
+
+  update-inventory:
+    required: false
+    default: false
+    description: Whether to send an update request
+
+  config-file-path:
+    required: false
+    default: './wss-unified-agent.config'
+    description: Configuration file name (including file path).
+
+  yarn-workspaces:
+    required: false
+    default: false
+    description: True if the project is structured as a monorepo using yarn workspaces
+
+  log-level:
+    required: false
+    default: info
+    description: Specifies the minimal log level printed to stdout/command line
+
+  locations-to-scan:
+    required: false
+    default: ./
+    description: Comma-separated list of directories and/or files to scan
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Install Java
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        java-version: '17'
+        java-package: jre
+
+    - name: Copy workspaces lock file to subfolders
+      if: ${{inputs.yarn-workspaces == 'true'}}
+      run: |
+          export ROOT_WORKSPACE=$(pwd)
+          yarn workspaces foreach exec '[ ! -f "./yarn.lock" ] && cp $ROOT_WORKSPACE/yarn.lock . || exit 0'
```

Mineures améliorations que je suggère ici:

- au lieu de "-f" qui ne fait que vérifier l'existence, c'est plus safe d'utiliser "-r" qui vérifie que l'utilisateur qui roule le script a le droit de lire le fichier (donc le cp a plus de chances de fonctionner)
- Vérifier si yarn.lock est lisable une seule fois
- exit 1 si la/les commandes échouent

```suggestion
          export ROOT_WORKSPACE=$(pwd)
          [ ! -r "./yarn.lock" ] && yarn workspaces foreach exec 'cp $ROOT_WORKSPACE/yarn.lock . || exit 1'
```

Je n'ai pas testé par contre si ça fonctionne vraiment

---

Après discussion sur :discord:, ma suggestion était mauvaise car il faut checker si le workspace a un fichier `yarn.lock` spécifique et copier celui du root seulement s'il n'y en a pas. Donc de mes suggestions, juste le `-r` est applicable.